com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java

```diff
@@ -180,28 +181,45 @@ public String parse(String str) {
         modules.clear();
         properties.clear();
 
-        parsedBuffer = new StringBuffer(str);
-
         // set up the lexer and parser
         // walk the generated parse tree from the
         // first Lua chunk
 
         LuaLexer lexer = new LuaLexer(CharStreams.fromString(str));
         tokenStream = new CommonTokenStream(lexer);
+        rewriter = new TokenStreamRewriter(tokenStream);
+        
+        // remove comments
+        tokenStream.fill();
+        for (Token token : tokenStream.getTokens()) {
+             if (token.getChannel() == LuaLexer.COMMENTS) {
+                int type = token.getType();
+                if (type == LuaLexer.LINE_COMMENT) {
+                    // Single line comment
+                    rewriter.replace(token, System.lineSeparator());
+                }
+                else if (type == LuaLexer.COMMENT) {
+                    // Multiline comment
+                    rewriter.replace(token, System.lineSeparator().repeat(token.getText().split("\r\n|\r|\n").length - 1));
+                }
+             }
+        }
         LuaParser parser = new LuaParser(tokenStream);
+
         ParseTreeWalker walker = new ParseTreeWalker();
         walker.walk(this, parser.chunk());
+        String resultText = rewriter.getText();
```

Comment:
It's better to include the first `rewriter.getText()` call in TimeProfiler.

From documentation about rewriter:
https://www.antlr.org/api/JavaTool/org/antlr/v4/runtime/TokenStreamRewriter.html

>You can insert stuff, replace, and delete chunks. Note that the operations are done lazily--only if you convert the buffer to a [String](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html?is-external=true) with [TokenStream.getText()](https://www.antlr.org/api/Java/org/antlr/v4/runtime/TokenStream.html#getText()). This is very efficient because you are not moving data around all the time. As the buffer of tokens is converted to strings, the [getText()](https://www.antlr.org/api/Java/org/antlr/v4/runtime/TokenStreamRewriter.html#getText()) method(s) scan the input token stream and check to see if there is an operation at the current index. If so, the operation is done and then normal [String](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html?is-external=true) rendering continues on the buffer. This is like having multiple Turing machine instruction streams (programs) operating on a single input tape. :)